src/index.ts

```diff
@@ -68,7 +80,28 @@ export * from './lib/utils/preconditions/containers/UserPermissionsPrecondition'
 export * from './lib/utils/preconditions/IPreconditionContainer';
 export * from './lib/utils/preconditions/PreconditionContainerArray';
 export * from './lib/utils/preconditions/PreconditionContainerSingle';
-export * as CorePreconditions from './preconditions/index';
+export { ApplicationCommandRegistries };
+
+export const CorePreconditions = {
+	ClientPermissions,
+	Cooldown,
+	DMOnly,
+	Enabled,
+	GuildNewsOnly,
+	GuildNewsThreadOnly,
+	GuildOnly,
+	GuildPrivateThreadOnly,
+	GuildPublicThreadOnly,
+	GuildTextOnly,
+	GuildThreadOnly,
+	NSFW,
+	UserPermissions
+};
+
+export namespace CorePreconditions {
+	export type UserPermissionsPreconditionContext = PermissionPreconditionContext;
+	export type CooldownContext = CooldownPreconditionContext;
+}
```

Comment:
Can't you just do this?
```typescript
export namespace CorePreconditions {
	export { CorePrecondition as ClientPermissions, type PermissionPreconditionContext } from './preconditions/ClientPermissions';
	export { CorePrecondition as Cooldown, type CooldownPreconditionContext } from './preconditions/Cooldown';
	// ...
}
```

Reply:
nope apparently not
![image](https://user-images.githubusercontent.com/4019718/150774777-16a51f94-5668-4f5c-8382-3669767909f8.png)

Reply:
Besides, namespaces are TS only.